src/commands/shell.mjs

```diff
@@ -0,0 +1,109 @@
+//@ts-check
+
+import repl from "node:repl";
+
+import { container } from "../cli.mjs";
+import {
+  // ensureDbScopeClient,
+  commonConfigurableQueryOptions,
+  getSimpleClient,
+} from "../lib/command-helpers.mjs";
+
+async function doShell(argv) {
+  const logger = container.resolve("logger");
+  let completionPromise;
+
+  if (argv.dbPath) logger.stdout(`Starting shell for database ${argv.dbPath}`);
+  logger.stdout("Type Ctrl+D or .exit to exit the shell");
+
+  /** @type {import('node:repl').ReplOptions} */
+  const replArgs = {
+    prompt: `${argv.db_path || ""}> `,
+    ignoreUndefined: true,
+    preview: argv.version !== "10",
+    // TODO: integrate with fql-analyzer for completions
+    completer: argv.version === "10" ? () => [] : undefined,
+    output: container.resolve("stdoutStream"),
+    input: container.resolve("stdinStream"),
+    eval: await buildCustomEval(argv),
+    terminal: true,
+  };
+
+  const shell = repl.start(replArgs);
```

Comment:
like the prior CLI, this uses node's REPL, which gives us lots of thing for free (syntax highlighting in some cases, standard keyboard shortcuts, prompts, etc).